lib/trino-parquet/src/main/java/io/trino/parquet/reader/ShortDecimalColumnReader.java

```diff
@@ -147,4 +162,21 @@ else if (columnDescriptor.getPrimitiveType().getPrimitiveTypeName() == INT64) {
             }
         }
     }
+
+    private void checkBytesFitInShortDecimal(byte[] bytes, int endOffset, Type trinoType)
+    {
+        byte expectedValue = 0;
+        if (bytes[endOffset] < 0) {
```

Comment:
There's a trick to do this in a branchless manner:

`bytes[endOffset] >> 7`

If the value is is positive, it produces a 0. If it's negative, it produces a -1 by extending the sign to all the lower bits.

Reply:
nice, updated to that now